### PR TITLE
Remove parentheses from exception names

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: src/transform.ts [hunk 1]
```diff
@@ -46,6 +46,22 @@ export function addCoersion(def: any): void {
   }
 }
 
+export function findRefs(definition: any): string[] {
```
Comment (arielshaqed) [inline src/transform.ts]: This seems to be the application of some function `ref` to a higher-order function, e.g. `deepFlatMap`.  I think writing it that way may make it easier to understand.   You might even find a suitable HOF in Ramda.

File: src/transform.ts [hunk 2]
```diff
@@ -46,6 +46,22 @@ export function addCoersion(def: any): void {
...
+  if (isPlainObject(definition)) {
+    const refs = flatMap(Object.values(definition), findRefs);
+    if (definition.$ref) {
+      return [definition.$ref, ...refs];
```
Comment (arielshaqed) [inline src/transform.ts]: Wouldn't `push`ing it at the end be considerably more efficient?

File: src/transform.ts [hunk 3]
```diff
@@ -226,7 +242,7 @@ export function transformClassPair(
           last: i === params.length - 1,
         })),
         returnType: typeToString(method.properties.returns).replace(/^null$/, 'void'),
-        throws: method.properties.throws ? typeToString(method.properties.throws).split(' | ') : [],
+        throws: findRefs(method.properties.throws).map(deref),
```
Comment (arielshaqed) [inline src/transform.ts:250]: I don't understand `findRefs` here.  E.g. if it throws
```ts
class DidntThrowFooError {
  // ...
  public notThrown: FooError;
}
```
then AFAICT this says it throws a `FooError`.

**Review by arielshaqed — COMMENTED** [review]
Doesn't seem right (or maybe I don't understand how you walk the JSON Schema).

File: src/transform.ts (hunk 2 above)
Comment (bergundy) [reply src/transform.ts]: Yes, but will also mutate the original array.

File: src/transform.ts (hunk 1 above)
Comment (bergundy) [reply src/transform.ts]: Probably need a `deepFind`, I'll write one.

File: src/transform.ts (hunk 3 above)
Comment (arielshaqed) [reply src/transform.ts:250]: Cool, the comment explains it!

**Review by arielshaqed — APPROVED** [review]
Great, thanks!

**Pull request merged**